miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/DefaultAuthorizationManager.java

```diff
@@ -0,0 +1,134 @@
+package uk.ac.bbsrc.tgac.miso.service.security;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.context.SecurityContextHolderStrategy;
+
+import com.eaglegenomics.simlims.core.User;
+import com.eaglegenomics.simlims.core.manager.SecurityManager;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.UserImpl;
+import uk.ac.bbsrc.tgac.miso.core.security.SecurableByProfile;
+import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
+
+public class DefaultAuthorizationManager implements AuthorizationManager {
+  
+  private static final Logger log = LoggerFactory.getLogger(DefaultAuthorizationManager.class);
+  
+  public static final String UNKNOWN_USER = "Unknown";
+  
+  @Autowired
+  private SecurityManager securityManager;
+  
+  private final SecurityContextHolderStrategy securityContextHolderStrategy;
+  
+  public DefaultAuthorizationManager() {
+    this.securityContextHolderStrategy = SecurityContextHolder.getContextHolderStrategy();
+  }
+  
+  @Override
+  public User getCurrentUser() throws IOException {
+    Authentication auth = securityContextHolderStrategy.getContext().getAuthentication();
+    if (auth == null) {
+      return null;
+    }
+    User user = securityManager.getUserByLoginName(auth.getName());
+    if (user == null && auth.isAuthenticated()) {
+      user = new UserImpl();
+      user.setAdmin(true);
```

Comment:
There's no way to do an admin check at this point. I'm not really sure of the case that this supports. The method was copied from UserAuthMisoRequestManager